File: 10-days-of-statistics/day-0-mean-median-and-mode/Solution.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int size = s.nextInt();
        int [] a = new int[size]; 
        for(int i = 0;i < size;i++) {
            a[i] = s.nextInt();
        }
        System.out.println(getMean(a));
        System.out.println(getMedian(a));
        System.out.println(getMode(a));
    }
    
    private static double getMean(int[] a) {
        double mean = 0;
        int sum = 0;
        int size = a.length;
        for(int i : a) {
            sum += i;
        }
        mean = (double) sum/size;
        return mean;
    }
    
    private static double getMedian(int[] a) {
        double median = 0;
        int size = a.length;
        int [] copy = a.clone();
        Arrays.sort(copy);
        if(size % 2 == 0)
            median = (double) (copy[size/2 - 1] + copy[size/2]) / 2;
        else {
            median = (double) copy[(size-1)/2];
        }
        return median;
    }
    
    private static int getMode(int[] a) {
        int mode = 0;
        int size = a.length;
        int [] copy = a.clone();
        Arrays.sort(copy);
        int count = 0, max = 0;
        int current = copy[0];
        for(int i = 0;i < size;i++) {
            if (copy[i] == current) {
                count++;
            } else {
                count = 1;
                current = copy[i];
            }
            if (count > max) {
                max = count;
                mode = copy[i];
            }
        }
        return mode;
    }
}

File: 10-days-of-statistics/day-0-mean-median-and-mode/Solution.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

double getMean(const vector<int> &array)
{
    int sum = 0;
    for (int i = 0; i < array.size(); ++i)
        sum += array[i];
    double mean = (double)(sum) / array.size();
    return mean;
}

double getMedian(const vector<int> &array) {
    double median = 0;
    int size = array.size();
    vector<int> copy (array.size());
    copy = array;
    sort (copy.begin(), copy.end());  
    if(size % 2 == 0)
        median = (double) (copy[size/2 - 1] + copy[size/2]) / 2;
    else {
        median = (double) copy[(size-1)/2];
    }
        return median;
}

int getMode(const vector<int> &array) {
    int mode = 0;
    int size = array.size();
    int count = 0, max = 0;
    vector<int> copy (array.size());
    copy = array;
    sort (copy.begin(), copy.end());  
    int current = 0;
    for(int i = 0;i < size;i++) {
       if (copy[i] == current) {
           count++;
       } else {
           count = 1;
           current = copy[i];
       }
       if (count > max) {
           max = count;
           mode = copy[i];
       }
     }
     return mode;
}

int main() {
    int size;
    cin >> size;
    vector<int> a(size);
    for (int i = 0;i < size;i++) 
        cin >> a[i];
    cout << getMean(a) << endl;
    cout << getMedian(a) << endl;
    cout << getMode(a) << endl;
    return 0;
}